core/commands/pubsub.go

```diff
@@ -118,16 +122,33 @@ To use, the daemon must be run with '--enable-pubsub-experiment'.
 	},
 	Marshalers: cmds.MarshalerMap{
 		cmds.Text: getPsMsgMarshaler(func(m *floodsub.Message) (io.Reader, error) {
+			if m.Message == nil {
```

Comment:
You can leave this out too, passing a `nil` byte array to a `bytes.NewReader` will result in the same as passing the empty string to a stringsreader